### ISSUE TITLE
Thorough review and update existing deployment instructions

### ISSUE THREAD

**Issue by gregelin** [GovReady/govready-q#581]
We have documentation for various installation and deployment. Documentation s old and needs to be reviewed, updated or retired as appropriate.

---
Components to this task (edit by Ethan):

- [x] Split off OS-specific and application-specific instructions
- [x] Write doc page for prod vs. dev
- [ ] Pull out needed details from Docker page
  - [ ] mention automated tests on the dev deployment page
  - [x] figure out this mailgun thing? or create a separate issue for it
- [ ] Check and update prose to account for reorganization (e.g., explain why Postgres section is `yum`-heavy, OR make it more neutral on distro)

**Comment by ethankaminski** [comment]
Three things I needed to install, which weren't mentioned, were the `Python.h` file (should be mentioned), `gcc` (should be mentioned, although I expected this would be needed), and `mysql_config` (covered by issue #505). That's something to include as part of the documentation update.

**Comment by ethankaminski** [comment]
Oh right, and other things I've got in my notes:

* It would be good to provide `pip3 install --user` as the snippet, because this seems to be recommended (by a first-party warning message) as better than `sudo`
* There are at least 3 sets of Linux instructions, so it would be good to check those for unintended discrepancies

**Comment by ethankaminski** [comment]
Note from a call: part of this will be explaining weird "gotchas" and such (one oddity that pops out to me is why `xvfb` is required - it's a Debian workaround).

**Comment by gregelin** [comment]
@ethankaminski @peterkaminski I am wondering if we should begin to split out instructions by OS?

**Comment by ethankaminski** [comment]
@gregelin Yeah, splitting the instructions in a more logical/deliberate fashion is something that struck me as a good idea, after I combed through the docs yesterday to get a sense for what the current state is.

Another thing that might seems worthwhile is organizing the docs a bit more by task, and also bringing the OS pages closer into sync. Currently, I noticed that each OS's page is opinionated about stuff like which public-facing webserver is used (_differently_ opinionated, even), which seems a bit odd.

An approach I was considering is spliting off things like DB-specific configuration or webserver-specific  configuration(Apache/Nginx) into their own pages, where possible, and then referencing those from the appropriate OS page.

Also, FWIW, I won't be able to test Mac or Windows instructions, but I can do most flavours of Linux pretty readily.

**Comment by gregelin** [comment]
@ethankaminski That all makes sense. Maybe propose an outline within this thread, or in a linked wiki page and we can discuss.

I think Windows deployments need to be via Docker containers for time being. @peterkaminski  and I can test Mac instructions.

**Comment by ethankaminski** [comment]
@gregelin @peterkaminski Proposed restructure at https://github.com/GovReady/govready-q/wiki/Deployment-Documentation-proposed-structure.

I might clean that up a bit more tomorrow, but it has all the essentials (both in terms of proposed structure, and in terms of addressing various notes I took while reviewing the deployment docs).

**Comment by gregelin** [comment]
Read-through the page. Organization looks good.

It might be helpful to see the install sections outlined within the overall documentation to see flows. 

We should also have sections to address “Enterprise Installations.”

**Comment by ethankaminski** [comment]
Oh, yeah, having an overview does make sense. Maybe on the Deployment section's landing/intro page?

Enterprise installation is a good idea, though I'm not familiar enough with Q to write that. In terms of where to put it, that seems quite similar to production/development, so that section seems good. I'll have to think about what to name that section, too. Maybe something with "Environments" in the page title, and maybe rename the current "Environment Settings" page to "Application Settings (environment.json)" or similar.

Come to think of it, moving the prod/dev/enterprise instructions to _before_ DB/webserver instructions would make sense. I may play around with the section ordering to see what feels best.

**Comment by ethankaminski** [comment]
Oh wait, would the enterprise installation docs just comprise [that one section in "Environment Settings"](https://govready-q.readthedocs.io/en/latest/Environment.html#enterprise-login)? Or is there more to it?

**Comment by gregelin** [comment]
I think Enterprise Installations should be its own major section. I've
already started that with https://github.com/GovReady/govready-q/pull/600

Part of this is to address in place items like single-sign on, but also to
talk about how enterprises operate GovReady-Q and manage the authoring of
compliance applications.

Greg Elin
CEO, GovReady PBC
p: 917-304-3488
e: gregelin@govready.com

**Comment by ethankaminski** [comment]
Oh, I should probably branch off of that, rather than off of master, then.

**Comment by gregelin** [comment]
That would make sense.

**Comment by ethankaminski** [comment]
I've pushed a [docs-restructure](https://github.com/GovReady/govready-q/tree/docs-restructure) branch with my current changes. Not sure how frequently I should open pull requests. Nightly, maybe? (every intermediate step should still be sane when compared to the previous state, so it largely depends on how frequently you'd prefer to have things synced into master)

Also, on a related note, it looks like there's an issue (#602) with how we've got `toctree` and `maxdepth` set up. I was hoping I'd be able to set a shallower maxdepth on some of these pages, but #602 prevents that. Mainly a cosmetic/QoL issue.
